common/lanelet2_extension/lib/CarmaTrafficSignal.cpp

```diff
@@ -60,6 +60,42 @@ LineStrings3d CarmaTrafficSignal::stopLine()
   return getParameters<LineString3d>(RoleName::RefLine);
 }
 
+Optional<ConstLineString3d> CarmaTrafficSignal::getStopLine(const ConstLanelet& llt) const 
+{
+  auto sl = stopLine();
+  if (sl.empty()) {
+    return {};
+  }
+  auto llts = getControlStartLanelets();
+  auto it = std::find(llts.begin(), llts.end(), llt);
+  if (llts.empty())
+  {
+    return {};
+  }
+  if (it == llts.end()) {
+    return {};
+  }
+  return sl.at(size_t(std::distance(llts.begin(), it)));
+}
+
+Optional<LineString3d> CarmaTrafficSignal::getStopLine(const ConstLanelet& llt) 
+{
+  auto sl = stopLine();
+  if (sl.empty()) {
+    return {};
+  }
+  lanelet::ConstLanelets llts = getControlStartLanelets();
+  if (llts.empty())
+  {
+    return {};
+  }
+  auto it = std::find(llts.begin(), llts.end(), llt);
+  if (it == llts.end()) {
+    return {};
+  }
+  return sl.at(size_t(std::distance(llts.begin(), it)));
+}
```

Comment:
This function contains an identical bit of logic as the const get stop line function. 
Is it possible for this function to be the main one and the const function simply casts the output?

Reply:
It looks like I can't use the main function inside a function with `const` identifier. Also the function overload seems to not work that way. So I just renamed one to have Const in the name, and edited as you suggested